In [1]:
import pandas as pd
import warnings as wr
wr.filterwarnings('ignore')

In [45]:
df = pd.read_csv('weight-height.csv')
df.head()


Gender     Height      Weight
0   Male  73.847017  241.893563
1   Male  68.781904  162.310473
2   Male  74.110105  212.740856
3   Male  71.730978  220.042470
4   Male  69.881796  206.349801

In [46]:
df1 = df.copy()

In [3]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()

In [4]:
df.Gender = le.fit_transform(df[['Gender']])

In [5]:
df.head()

Gender     Height      Weight
0       1  73.847017  241.893563
1       1  68.781904  162.310473
2       1  74.110105  212.740856
3       1  71.730978  220.042470
4       1  69.881796  206.349801

In [6]:
df.tail()

Gender     Height      Weight
8550       0  60.483946  110.565497
8551       0  63.423372  129.921671
8552       0  65.584057  155.942671
8553       0  67.429971  151.678405
8554       0  60.921791  131.253738

### Going  ahead by separating data 


In [7]:
x = df.drop("Weight", axis = 1)
x

Gender     Height
0          1  73.847017
1          1  68.781904
2          1  74.110105
3          1  71.730978
4          1  69.881796
...      ...        ...
8550       0  60.483946
8551       0  63.423372
8552       0  65.584057
8553       0  67.429971
8554       0  60.921791

[8555 rows x 2 columns]

In [8]:
y = df['Weight']
y 

0       241.893563
1       162.310473
2       212.740856
3       220.042470
4       206.349801
           ...    
8550    110.565497
8551    129.921671
8552    155.942671
8553    151.678405
8554    131.253738
Name: Weight, Length: 8555, dtype: float64

In [9]:
import numpy as np
from sklearn.model_selection import train_test_split

In [10]:
x_train, x_test , y_train, y_test = train_test_split(x, y , test_size = .30, random_state = 12)

In [11]:
from sklearn.linear_model import LinearRegression

In [12]:
reg = LinearRegression()

In [13]:
x_train.shape

(5988, 2)

In [14]:
type(x_train)

pandas.core.frame.DataFrame

In [15]:
y_train.shape

(5988,)

In [16]:
reg.fit(x_train, y_train)

LinearRegression()

In [17]:
reg.predict(x_train)

array([177.75973175, 205.12220575, 195.96341247, ..., 148.77365458,
       150.88474836, 152.39845129])

In [18]:
reg.predict(x_test)

array([155.21990368, 166.36105918, 129.74602898, ..., 195.28641803,
       127.64543442, 175.03619195])

In [19]:
from sklearn.metrics import r2_score
r2_score(y_train,reg.predict(x_train))

0.8996883838293749

In [20]:
reg.score(x_test, y_test)

0.9005089705784229

In [21]:
from sklearn.metrics import mean_squared_error

In [22]:
mse  = mean_squared_error(y_train, reg.predict(x_train))

In [23]:
mse

103.59585759197246

### Using KNeighborsRegressor  method


In [24]:
from sklearn.neighbors import KNeighborsRegressor

In [25]:
neighbors = KNeighborsRegressor(n_neighbors = 5)

In [26]:
neighbors.fit(x_train, y_train)

KNeighborsRegressor()

In [27]:
pred_y = neighbors.predict(x_train)

In [28]:
r2_score(y_train, pred_y)

0.9191366690929216

In [29]:
mse_For_KNN = mean_squared_error(y_train,pred_y)

In [30]:
mse

103.59585759197246

#### Here r2 score for LinearRegression 0.8996883838293749
#### and r2 score for KNN 0.9191366690929216   

### Use KNeighbors Classifier

In [49]:
df1

Gender     Height      Weight
0       Male  73.847017  241.893563
1       Male  68.781904  162.310473
2       Male  74.110105  212.740856
3       Male  71.730978  220.042470
4       Male  69.881796  206.349801
...      ...        ...         ...
8550  Female  60.483946  110.565497
8551  Female  63.423372  129.921671
8552  Female  65.584057  155.942671
8553  Female  67.429971  151.678405
8554  Female  60.921791  131.253738

[8555 rows x 3 columns]

In [50]:
cls_x = df1.drop("Gender", axis = 1)
cls_x

Height      Weight
0     73.847017  241.893563
1     68.781904  162.310473
2     74.110105  212.740856
3     71.730978  220.042470
4     69.881796  206.349801
...         ...         ...
8550  60.483946  110.565497
8551  63.423372  129.921671
8552  65.584057  155.942671
8553  67.429971  151.678405
8554  60.921791  131.253738

[8555 rows x 2 columns]

In [51]:
cls_y = df1["Gender"]
cls_y

0         Male
1         Male
2         Male
3         Male
4         Male
         ...  
8550    Female
8551    Female
8552    Female
8553    Female
8554    Female
Name: Gender, Length: 8555, dtype: object

In [53]:
cls_x_train, cls_x_test, cls_y_train,cls_y_test = train_test_split(cls_x, cls_y, test_size =.30 , random_state = 42)

In [55]:
cls_x_train

Height      Weight
553   67.594031  186.751417
1397  71.601697  211.031652
7934  62.625985  143.768451
8367  66.136131  151.814648
3320  71.843308  196.505814
...         ...         ...
5734  63.349988  139.223151
5191  58.861625   90.728561
5390  64.325406  156.800504
860   69.074492  179.386817
7270  67.588840  150.380597

[5988 rows x 2 columns]

In [57]:
from sklearn.neighbors import KNeighborsClassifier

In [64]:
cls_neigh  = KNeighborsClassifier(n_neighbors = 5, p =2)

In [65]:
cls_neigh.fit(cls_x_train, cls_y_train)

KNeighborsClassifier()

In [66]:
cls_pred_y = cls_neigh.predict(cls_x_train)

In [67]:
cls_pred_y

array(['Male', 'Male', 'Female', ..., 'Male', 'Male', 'Female'],
      dtype=object)

In [70]:
from sklearn.metrics import accuracy_score
accuracy_score(cls_y_train, cls_pred_y)

0.9291917167668671